module/VuFind/src/VuFind/Session/SecureDelegatorFactory.php

```diff
@@ -0,0 +1,66 @@
+<?php
+/**
+ * Secure session delegator factory
+ *
+ * Copyright (C) Villanova University 2018,
+ *               Leipzig University Library <info@ub.uni-leipzig.de> 2018.
+ *
+ * PHP version 7
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 2,
+ * as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ *
+ * @category VuFind
+ * @package  Session_Handlers
+ * @author   Demian Katz <demian.katz@villanova.edu>
+ * @author   Sebastian Kehr <kehr@ub.uni-leipzig.de>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org/wiki/development:plugins:session_handlers Wiki
+ */
+namespace VuFind\Session;
+
+use Interop\Container\ContainerInterface;
+use Zend\ServiceManager\Factory\DelegatorFactoryInterface;
+
+/**
+ * Secure session delegator factory
+ *
+ * @category VuFind
+ * @package  Session_Handlers
+ * @author   Demian Katz <demian.katz@villanova.edu>
+ * @author   Sebastian Kehr <kehr@ub.uni-leipzig.de>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org/wiki/development:plugins:session_handlers Wiki
+ */
+class SecureDelegatorFactory implements DelegatorFactoryInterface
+{
+    /**
+     * Invokes this factory.
+     *
+     * @param ContainerInterface $container Service container
+     * @param string             $name      Service name
+     * @param callable           $callback  Service callback
+     * @param array|null         $options   Service options
+     *
+     * @return SecureDelegator
+     */
+    public function __invoke(
+        ContainerInterface $container, $name, callable $callback,
+        array $options = null
+    ) {
+        $handler = call_user_func($callback);
+        $config = $container->get('VuFind\Config\PluginManager');
+        $secure = $config->get('config')->Session->secure;
```

Comment:
I'd recommend setting a default here in case the configuration is unset (i.e. `$config->get('config')->Session->secure ?? false`).

Reply:
It also looks like the changes to config.ini to add this setting are not included in this PR.

Reply:
> I'd recommend setting a default here

Why not, though `null` gets converted to boolean `false` anyway.

> changes to config.ini to add this setting are not included

Now included with `false` being default.

Reply:
My main concern regarding setting a default is avoiding notices being thrown about undefined variables -- though perhaps in this context that is unlikely to happen; I can't remember off the top of my head under what circumstances, if any, `Zend\Config\Config` will trigger such notices.